Mathlib/NumberTheory/SmoothNumbers.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2023 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.LinearAlgebra.ExteriorAlgebra.Grading
+import Mathlib.Data.Nat.Factorization.Basic
+
+/-!
+# Smooth numbers
+
+We define the set `Nat.smoothNumbers n` consisting of the positive natural numbers all of
+whose prime factors are strictly less than `n`.
+
+We also define the finite set `Nat.primesBelow n` to be the set of prime numbers less than `n`.
+
+The main definition `Nat.equivProdNatSmoothNumbers` establishes the bijection between
+`ℕ × (smoothNumbers p)` and `smoothNumbers (p+1)` given by sending `(e, n)` to `p^e * n`.
+Here `p` is a prime number.
+-/
+
+namespace Nat
+
+/-- `primesBelow n` is the set of primes less than `n` as a finset. -/
+def primesBelow (n : ℕ) : Finset ℕ := (Finset.range n).filter (fun p ↦ p.Prime)
+
+@[simp]
+lemma primesBelow_zero : primesBelow 0 = ∅ := rfl
```

Comment:
This is true definitionally, but I'm not sure that's guaranteed. I'd be inclined to use the API with
```suggestion
lemma primesBelow_zero : primesBelow 0 = ∅ := by
  rw [primesBelow, Finset.range_zero, Finset.filter_empty]
```

Reply:
OK.